File: MultiLanguageSandbox/src/test/deps/elixir/9.exs
defmodule JumpGame do
def can_reach_end?(nums) do
    can_reach_end?(nums, 0, 0)
  end

  defp can_reach_end?(_nums, current_index, max_reach) when current_index > max_reach, do: false
  defp can_reach_end?(_nums, current_index, _max_reach) when current_index == length(_nums) - 1, do: true
  defp can_reach_end?(nums, current_index, max_reach) do
    new_max_reach = max(max_reach, current_index + Enum.at(nums, current_index))
    can_reach_end?(nums, current_index + 1, new_max_reach)
  end
end



ExUnit.start()
defmodule JumpGameTest do
use ExUnit.Case

test "can reach the end" do
assert JumpGame.can_reach_end?([2, 3, 1, 1, 4])
refute JumpGame.can_reach_end?([3, 2, 1, 0, 4])
end

test "handles single element array" do
assert JumpGame.can_reach_end?([0])
end

test "handles empty array" do
refute JumpGame.can_reach_end?([])
end
end

File: MultiLanguageSandbox/src/test/deps/swift/9.swift
func countDivisiblePairs(_ nums: [Int], _ k: Int) -> Int {
    var count = 0
    let n = nums.count
    
    for i in 0..<n {
        for j in i+1..<n {
            if nums[i] == nums[j] && (i * j) % k == 0 {
                count += 1
            }
        }
    }
    
    return count
}


func checkCountDivisiblePairs(_ countDivisiblePairs: ([Int], Int) -> Int) {
assert(countDivisiblePairs([1, 2, 3, 1, 2, 3], 3) == 1)
assert(countDivisiblePairs([1, 1, 1, 1], 1) == 6)
assert(countDivisiblePairs([1, 2, 3, 4, 5, 6], 7) == 0)
assert(countDivisiblePairs([1, 2, 2, 1, 2], 2) == 4)
assert(countDivisiblePairs([4, 4, 4, 4], 4) == 3)
assert(countDivisiblePairs([], 5) == 0)
assert(countDivisiblePairs([2, 2, 2, 2, 2], 3) == 7 )
}

checkCountDivisiblePairs(countDivisiblePairs)